### DISCUSSION TITLE
No Access Token when running inside cluster?

### DISCUSSION THREAD

**Discussion by HenrikDK** [kubernetes-client/csharp#863] [Q&A] [answered]
Hi 

I'm using the C# client to fetch information from api server. 

I decided to only partially use the client library to fetch "access token" that you need to attach as a bearer when making api requests.

While the approach worked fine while testing locally all requests started failing when the pod ran inside the cluster.

I added some logging so I could see what's happening inside the cluster: 

```
if (KubernetesClientConfiguration.IsInCluster()){
    Log.Logger.Information($"Controller running in cluster with host \"{config.Host}\" using access token \"{config.AccessToken}\"");
    Log.Logger.Information(JsonConvert.SerializeObject(config));
}
```

When I run my pod inside a cluster the access token property is empty. And I get following message: 

```
{"Timestamp":"2022-05-27T11:05:38.3458927+00:00","Level":"Information","MessageTemplate":"Controller running in cluster with host \"https://172.18.0.1:443/\" using access token \"\""}
```

Does the client use some other authentication method when running inside the cluster?

**Comment by tg123** [comment, marked as answer]
you can get token from `/var/run/secrets/kubernetes.io/serviceaccount/token`

the client will load token from the file above automatically 
the config.token is for you to see static token and will not be populated


free free to reopen if you still have any questions

**Comment by HenrikDK** [comment]
Thanks for the timely clarification :)

Guess I was expecting something different from a "config" object.

**Comment by sfahadrizvi** [comment]
This item is closed but for future users who have this issue https://github.com/kubernetes-client/csharp/issues/1556#issuecomment-2103450570
